### PR TITLE
chatd: init at 1.1.2

### PR DESCRIPTION
**Pull request by lucasew**
## Description of changes

Replaces #291039

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 325099` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>chatd</li>
  </ul>
</details>

File: pkgs/by-name/ch/chatd/package.nix [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  postInstall = ''
+    substituteInPlace $out/share/chatd/node_modules/@xenova/transformers/src/env.js \
+      --replace-fail "import fs from 'fs';" "import fs from 'fs';import os from 'os';" \
+      --replace-fail 'path.dirname(path.dirname(url.fileURLToPath(import.meta.url)))' 'path.join(os.homedir(), ".cache", "chatd")'
+
+    rm -rf $out/share/electron{,-winstaller} $(find $out -name 'win32')
+    find $out/share/chatd/node_modules -name '*.exe' -delete
+    find $out/share/chatd/node_modules -name '*.dll' -delete
+    find $out/share/chatd/node_modules -name '*.pdb' -delete
+    find $out/share/chatd/node_modules -name '*.pdb' -delete
```
Comment (SuperSandro2000) [inline pkgs/by-name/ch/chatd/package.nix:69]:
```suggestion
    find $out/share/chatd/node_modules -name '*.exe' -or -name '*.dll' -or -name '*.pdb' -delete
```
the pdf line was diplicated, right?

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 325099` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/by-name/ch/chatd/package.nix [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    rm -rf ${lib.concatStringsSep " " ([]
+    ++ (lib.optional (!stdenv.isx86_64) "$out/share/chatd/node_modules/onnxruntime-node/bin/napi-v3/*/x64")
+    ++ (lib.optional (!stdenv.isAarch64) "$out/share/chatd/node_modules/onnxruntime-node/bin/napi-v3/*/arm64")
+    ++ (lib.optional (!stdenv.isDarwin) "$out/share/chatd/node_modules/onnxruntime-node/bin/napi-v3/darwin")
+    ++ (lib.optional (!stdenv.isLinux) "$out/share/chatd/node_modules/onnxruntime-node/bin/napi-v3/linux")
+    )}
```
Comment (SuperSandro2000) [inline pkgs/by-name/ch/chatd/package.nix:75]:
```suggestion
    rm -rf ${lib.concatStringsSep " " (
      lib.optional (!stdenv.isx86_64) "$out/share/chatd/node_modules/onnxruntime-node/bin/napi-v3/*/x64"
      ++ lib.optional (!stdenv.isAarch64) "$out/share/chatd/node_modules/onnxruntime-node/bin/napi-v3/*/arm64"
      ++ lib.optional (!stdenv.isDarwin) "$out/share/chatd/node_modules/onnxruntime-node/bin/napi-v3/darwin"
      ++ lib.optional (!stdenv.isLinux) "$out/share/chatd/node_modules/onnxruntime-node/bin/napi-v3/linux"
    )}
```

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM

File: pkgs/by-name/ch/chatd/package.nix [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  '';
+
+  passthru.updateScript = gitUpdater { rev-prefix = "v"; };
+
+  meta = with lib; {
```
Comment (superherointj) [inline pkgs/by-name/ch/chatd/package.nix:80]:
```suggestion
  meta = {
```

`meta.changelog` is missing

> changelog = "https://github.com/BruceMacD/chatd/releases/tag/v${version}";

File: pkgs/by-name/ch/chatd/package.nix [hunk 4]
```diff
@@ -0,0 +1,88 @@
+{ buildNpmPackage
```
Comment (superherointj) [inline pkgs/by-name/ch/chatd/package.nix:1]: Use `nixfmt` (rfc)

Comment (lucasew) [reply pkgs/by-name/ch/chatd/package.nix:1]: nixpkgs-fmt on the file is doing a no-op, so it seems it's right

File: pkgs/by-name/ch/chatd/package.nix (hunk 3 above)
Comment (lucasew) [reply pkgs/by-name/ch/chatd/package.nix]: done

File: pkgs/by-name/ch/chatd/package.nix (hunk 4 above)
Comment (lucasew) [reply pkgs/by-name/ch/chatd/package.nix]: found out how to do that, now it's fixed

File: pkgs/by-name/ch/chatd/package.nix (hunk 2 above)
Comment (lucasew) [reply pkgs/by-name/ch/chatd/package.nix]: done, but the formatter changed this part

File: pkgs/by-name/ch/chatd/package.nix (hunk 1 above)
Comment (lucasew) [reply pkgs/by-name/ch/chatd/package.nix]: I think it was, but not anymore

**Comment by superherointj** [comment]
Git commit is missing release:

> Release: https://github.com/BruceMacD/chatd/releases/tag/v1.1.2

**Review by superherointj — APPROVED** [review]

**Pull request merged**